lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.ICU.cpp

```diff
@@ -224,55 +227,44 @@ namespace PlatformAgnostic
             return u_isUWhiteSpace(ch) == 1;
         }
 
-        int32 ChangeStringLinguisticCase(CaseFlags caseFlags, const char16* sourceString, uint32 sourceLength, char16* destString, uint32 destLength, ApiError* pErrorOut)
+        template<bool toUpper, bool useInvariant>
+        charcount_t ChangeStringLinguisticCase(const char16* sourceString, charcount_t sourceLength, char16* destString, charcount_t destLength, ApiError* pErrorOut)
         {
+            Assert(sourceString != nullptr && sourceLength > 0);
+            Assert(destString != nullptr || destLength == 0);
+
             int32_t resultStringLength = 0;
             UErrorCode errorCode = U_ZERO_ERROR;
+            *pErrorOut = ApiError::NoError;
 
-            if (caseFlags == CaseFlagsUpper)
+            // u_strTo treats nullptr as the system default locale and "" as root
+            const char* locale = useInvariant ? "" : nullptr;
+
+            if (toUpper)
             {
                 resultStringLength = u_strToUpper((UChar*) destString, destLength,
-                    (UChar*) sourceString, sourceLength, NULL, &errorCode);
-            }
-            else if (caseFlags == CaseFlagsLower)
-            {
-                resultStringLength = u_strToLower((UChar*) destString, destLength,
-                    (UChar*) sourceString, sourceLength, NULL, &errorCode);
+                    (UChar*) sourceString, sourceLength, locale, &errorCode);
             }
             else
             {
-                Assert(false);
+                resultStringLength = u_strToLower((UChar*) destString, destLength,
+                    (UChar*) sourceString, sourceLength, locale, &errorCode);
             }
 
-            if (U_FAILURE(errorCode) &&
-                !(destLength == 0 && errorCode == U_BUFFER_OVERFLOW_ERROR))
-            {
-                *pErrorOut = TranslateUErrorCode(errorCode);
-                return -1;
-            }
+            AssertMsg(resultStringLength > 0, "u_strToCase must return required destString length");
```

Comment:
Is this assert hit if you try to change the case of the empty string?

Reply:
Oh nevermind, I see there's a check earlier on that the source length is greater than 0; I'm not sure I see where the caller enforces that though